CMakeLists.txt

```diff
@@ -202,6 +202,22 @@ endif(ENABLE_ORC)
 ########################################################################
 add_subdirectory(docs)
 
+########################################################################
+# Setup Coverage
+########################################################################
+if(${CMAKE_BUILD_TYPE} STREQUAL "Coverage")
+    # Use CodeCoverage based on Lars Bilke's
+    # https://github.com/bilke/cmake-modules , with extensive
+    # modifications as done in the past for GNU Radio.
+    include(CodeCoverage)
+    append_coverage_compiler_flags()
+    setup_target_for_coverage_gcovr_html(
```

Comment:
Why only html output and why gcovr? IIRC lcov is the more mature and complete tool. Allowing regular coverage output we can hook that into codecov.io pipelines and add information to pullrequests and pushes to track coverage as we go.

Reply:
gcov vs lcov:

TBH, because I looked into lcov first, couldn't make it work, and then resorted to doing what was possible within limited time :)

html:

because I wanted this to be immediately useful for insights, and hence jumped on a minimum effort way of getting readable output. Proper CI integration, including things like complaining when you add code that is specifically not well-covered, is something that needs to come – but having this seemed better than not having it.

Reply:
gcov vs lcov:

TBH, because I looked into lcov first, couldn't make it work, and then resorted to doing what was possible within limited time :)

html:

because I wanted this to be immediately useful for insights, and hence jumped on a minimum effort way of getting readable output. Proper CI integration, including things like complaining when you add code that is specifically not well-covered, is something that needs to come – but having this seemed better than not having it.